### ISSUE TITLE
B3.a - Streaming - Végétation - données 2D

### ISSUE THREAD

**Issue by cbriand** [sigrennesmetropole/api-3D#32]
En tant qu'utilisateur de l'API 3D, je souhaite télécharger les données végétation. 

Pour valider cette API, je dois pouvoir me connecter au flux correspondant obtenir données 2D de végétation. 
Les trois niveaux de filtre géographique suivants doivent être utilisables : Bbox, commune, quartier.

**Comment by crenny35** [comment]
Avons nous les entrants pour commencer cette tache ?

**Comment by cbriand** [comment]
Non, le service WFS n'existe pas encore. Je relance mes collègues sur ce point.

**Comment by cbriand** [comment]
Le flux WFS est décrit sur la page de métadonnées suivante : https://public.sig.rennesmetropole.fr/geonetwork/srv/fre/catalog.search#/metadata/1acd23ad-c4b1-4ccb-83c6-33e6ef2214e9

**Comment by anthalcgi** [comment]
Pour le filtre commune, faut-il faire comme pour le mnt et utiliser la bbox de la commune ou bien utiliser les véritables limites de la commune ?

**Comment by cbriand** [comment]
l'API3D ne connait pas les vraies limites de communes. Pour faire simple, on retire le filtre sur la commune qui n'est pas vraiment pertinent. Par contre, on conserve le filtre sur bbox limité à 10km².

**Comment by cbriand** [comment]
Le service n'est pas fonctionnel et ne retourne aucun résultat. 
Le fichier rendu fait état d'une exception levée sur un paramètre invalide.

`<?xml version="1.0" encoding="UTF-8"?><ows:ExceptionReport xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ows="http://www.opengis.net/ows" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.0.0" xsi:schemaLocation="http://www.opengis.net/ows https://public.sig.rennesmetropole.fr/geoserver/schemas/ows/1.0.0/owsExceptionReport.xsd">
<ows:Exception exceptionCode="InvalidParameterValue" locator="typeName">
<ows:ExceptionText>Feature type :undefined unknown</ows:ExceptionText>
</ows:Exception>
</ows:ExceptionReport>`

**Comment by crenny35** [comment]
Es tu sur qu'il ne s'agissait pas d'une indisponibilité coté geoserver ? la requete suivante me renvoie des résultats : 
https://api3d.imfr.cgi.com/recette/api/collections/vegetation/items?bbox=1351534.8845,7223500.0490,1354849.5635,7225175.6015

**Comment by cbriand** [comment]
Non, ce n'est pas le geoserver. En effet, la requete passée depuis le serveur cgi retourne des éléments, et la même requête sur notre serveur retourne une erreur "No query specified".
https://api-3d.sig.rennesmetropole.fr/api/collections/vegetation/items?bbox=1351534.8845,7223500.0490,1354849.5635,7225175.6015

**Comment by cbriand** [comment]
Un configuration manquante sur notre serveur ?

**Comment by crenny35** [comment]
Est ce que tu as pris la derniere version du fichier deploy-env ? et ajouté les variables : 
<html>
<body>


COUCHE_ARBRES | Attribut couche envoyé a GeoServer pour la récupération de la végétation
-- | --
COUCHE_MOBILIER | Attribut couche envoyé a GeoServer pour la récupération du mobilier urbain


</body>
</html>

**Comment by cbriand** [comment]
Oui, le deploy.env est bine à jour

**Comment by cbriand** [comment]
Est-ce qu'il y aurait un moyen d'observer la requête WFS qui est appelée par l'API-3D pour voir ce qui cloche ?

**Comment by cbriand** [comment]
En effet, les deux variables dans les fichiers .env corrigent l'appel. Je qualifie les tickets demain.

**Comment by cbriand** [comment]
OK pour le service, par contre, il faudrait un nom de fichier plus parlant que wfs.json. Par exemple : vegetation.json

**Comment by cbriand** [comment]
OK, validé
